qiita_db/metadata_template.py

```diff
@@ -2232,19 +2234,30 @@ def load_template_to_dataframe(fn, strip_whitespace=True):
     |             longitude |        float |
     +-----------------------+--------------+
     """
-
-    # First, strip all values from the cells in the input file, if requested
+    # Load in file lines
+    holdfile = None
+    if isinstance(fn, str):
+        # filepath passed
+        with open(fn) as f:
+            holdfile = f.readlines()
+    else:
+        # open file passed
+        holdfile = fn.readlines()
+
+    # get and clean the required columns
+    reqcols = set(get_table_cols("required_sample_info"))
+    reqcols.add('sample_name')
+    reqcols.add('required_sample_info_status')
+    reqcols.discard('required_sample_info_status_id')
+    # clean all the column names
+    cols = holdfile[0].split('\t')
+    holdfile[0] = '\t'.join(c.lower() if c.lower() in reqcols else c
```

Comment:
You have to go through the strip_whitespace block before checking for this otherwise a string like `'lolercoaster '` would not be found, if `'lolercoaster'` was a required column.